web_widget_html_markdown/static/src/js/web_widget_html_markdown.js

```diff
@@ -0,0 +1,861 @@
+        description: _lt("HtmlMarkdown"),
+        classname: 'oe_form_field oe_form_field_html_markdown',
+        supportedFieldTypes: ['html'],
+
+        custom_events: {
+            wysiwyg_focus: '_onWysiwygFocus',
+            wysiwyg_blur: '_onWysiwygBlur',
+            wysiwyg_change: '_onChange',
+            wysiwyg_attachment: '_onAttachmentChange',
+        },
+
+        /**
+         * @override
+         */
+        willStart: function () {
+            var self = this;
+            this.isSwitch = false;
+            this.isRendered = false;
+            this._onUpdateIframeId = 'onLoad_' + _.uniqueId('FieldHtmlMarkdown');
+            var defAsset;
+            if (this.nodeOptions.cssReadonly) {
+                defAsset = ajax.loadAsset(this.nodeOptions.cssReadonly);
+            }
+
+            if (!assetsLoaded) { // Avoid flickering when begin to edit
+                assetsLoaded = new Promise(function (resolve) {
+                    var wysiwyg = new Wysiwyg(self, {});
+                    wysiwyg.attachTo($('<textarea>')).then(function () {
+                        wysiwyg.destroy();
+                        resolve();
+                    });
+                });
+            }
+
+            return Promise.all([this._super(), assetsLoaded, defAsset]);
+        },
+        /**
+         * @override
+         */
+        destroy: function () {
+            delete window.top[this._onUpdateIframeId];
+            if (this.$iframe) {
+                this.$iframe.remove();
+            }
+            this._super();
+        },
+
+        // --------------------------------------------------------------------------
+        // Public
+        // --------------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        activate: function (options) {
+            if (this.wysiwyg) {
+                this.wysiwyg.focus();
+                return true;
+            }
+        },
+        /**
+         * Wysiwyg doesn't notify for changes done in code mode. We override
+         * commitChanges to manually switch back to normal mode before committing
+         * changes, so that the widget is aware of the changes done in code mode.
+         *
+         * @override
+         */
+        commitChanges: function () {
+            var self = this;
+            if (config.isDebug() && this.mode === 'edit') {
+                var layoutInfo = $.summernote.core.dom.makeLayoutInfo(this.wysiwyg.$editor);
+                $.summernote.pluginEvents.codeview(undefined, undefined, layoutInfo, false);
+            }
+            if (self.mode == 'edit' && self.$is_markdown) {
+                // Transform in html before saving
+                return self.sw_mk_to_html().then(function() {
+                    self.is_markdown = false;
+                    // we use recursion (max1)!
+                    return self.commitChanges()
+                });
+            } else {
```

Comment:
Is this `else` correct? It would also trigger when `self.mode != 'edit'` and I don't think that's good... maybe an "else if self.mode == 'edit'"